src/explorer/SiteTreeItem.ts

```diff
@@ -109,19 +109,30 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
         const correlationId = getRandomHexString(10);
         telemetryProperties.correlationId = correlationId;
 
-        const workspaceConfig: WorkspaceConfiguration = workspace.getConfiguration(constants.extensionPrefix, Uri.file(fsPath));
+        const siteConfig: WebSiteModels.SiteConfigResource = await this.client.getSiteConfig();
+        let deploymentFilePath: string;
+        if (fsPath) {
+            deploymentFilePath = fsPath;
+        } else {
+            if (siteConfig.linuxFxVersion && siteConfig.linuxFxVersion.toLowerCase().startsWith('tomcat')) {
```

Comment:
Add 'tomcat' to the runtimes in constants.ts
https://github.com/Microsoft/vscode-azureappservice/blob/master/src/constants.ts